Add project's bluesky accounts to footer

karthikmanam opened the pull request:
<!--
 Hello!

 PLEASE title the FIRST commit appropriately, so that if you squash all
 your commits into one, the combined commit message makes sense.
 For overall help on editing and submitting pull requests, visit:
  https://kubernetes.io/docs/contribute/suggesting-improvements/

 Use the default base branch, “main”, if you're documenting existing
 features in the English localization.

 If you're working on a different localization (not English), see
 https://kubernetes.io/docs/contribute/new-content/overview/#choose-which-git-branch-to-use
 for advice.

 If you're documenting a feature that will be part of a future release, see
 https://kubernetes.io/docs/contribute/new-content/new-features/ for advice.
-->
### Description
Add project's bluesky accounts to footer
<!--
 Remember to ADD A DESCRIPTION and delete this note before submitting
 your pull request. The description should explain what will change,
 and why.
-->

### Issue
Adding bluesky accoutns to footer and removing twitter references
<!--
 If this pull request resolves an open issue, please link the issue in the PR
 description so it will automatically close when the PR is merged.

 See the GitHub documentation for more details and other options:

 https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword
-->

Closes: #49547

sftim commented on content/en/docs/reference/glossary/device-plugin.md:
```diff
@@ -12,7 +12,7 @@ tags:
  Device plugins run on worker
 {{< glossary_tooltip term_id="node" text="Nodes">}} and provide
-{{< glossary_tooltip term_id="pod" text="Pods ">}} with access to resources,
+{{< glossary_tooltip term_id="pod" text="Pods">}} with access to resources,
```
Why this change? I'd omit it.

sftim commented on hugo.toml:
```diff
@@ -263,6 +263,11 @@ no = 'Sorry to hear that. Please <a href="https://github.com/USERNAME/REPOSITORY
 # Replace with fa-x-twitter once available
 
+[[params.links.user]]
+  name_key = "community_bluesky_name"
```
I think you would need to add a localization string for this term.

karthikmanam replied:
@sftim Done

sftim replied:
I mean something like https://github.com/kubernetes/website/blob/3d243cd3949d3e024eec13c1fd01c0adf06db972/data/i18n/en/en.toml#L74 but for Bluesky

karthikmanam replied:
@sftim Done. Please review

sftim replied:
I don't see a change to `website/data/i18n/en/en.toml`

karthikmanam replied:
Now it should reflect @sftim

sftim commented on hugo.toml:
```diff
@@ -263,6 +263,11 @@ no = 'Sorry to hear that. Please <a href="https://github.com/USERNAME/REPOSITORY
 
+[[params.links.user]]
+  name_key = "community_bluesky_name"
+  url = "https://bsky.app/profile/kubernetes.bsky.social"
```
The link is to the wrong profile
```suggestion
  url = "https://bsky.app/profile/kubernetes.io"
```

Try this.

nate-double-u reviewed (commented):
Was a decision made to remove Twitter/X? Based on the conversations in the issue and linked Slack discussion, I'm not sure folks want it removed, just to have bluesky added.

/cc @kubernetes/contributor-comms

nate-double-u commented:
Related: https://github.com/kubernetes/website/pull/49606

/hold
/commitee steering

nate-double-u commented:
(helps if i spell it right, sorry for the noise 😅)

/committee steering

sftim commented:
> Steering has a majority consensus to support the comms team's decision here (that would be @kubernetes/contributor-comms, including @mfahlandt above).
> 
> For now then, please retain a full listing on the community page as suggested above, but we can rotate in bluesky in the footer and other places.

&mdash;https://github.com/kubernetes/website/pull/49606#issuecomment-2655084195

sftim commented:
Steering is happy to back the comms team decision: keep Twitter in the community page (we should make sure to add Bluesky there); it's OK to drop Twitter from the footer as comms team have this intent.

sftim commented on content/en/community/_index.html:
For this page, we should retain 𝕏 (formerly Twitter) and _add_ Bluesky.

Given that
/lgtm cancel

karthikmanam replied:
@sftim @nate-double-u I have made the appropriate change. PTAL and approve.

nate-double-u commented:
> Steering is happy to back the comms team decision: keep Twitter in the community page (we should make sure to add Bluesky there); it's OK to drop Twitter from the footer as comms team have this intent.

Thanks Tim, I hadn't seen that bit.

/hold cancel

abhilasha2418 commented:
@karthikmanam contributors are [advised](https://kubernetes.io/docs/contribute/new-content/open-a-pr/#squashing-commits) to squash their commits before a merge.

samsonkolge commented:
> @karthikmanam contributors are [advised](https://kubernetes.io/docs/contribute/new-content/open-a-pr/#squashing-commits) to squash their commits before a merge.

hi @abhilasha2418 I have now squashed the commits

karthikmanam commented:
> > @karthikmanam contributors are [advised](https://kubernetes.io/docs/contribute/new-content/open-a-pr/#squashing-commits) to squash their commits before a merge.
> 
> hi @abhilasha2418 I have now squashed the commits

@sftim Could you please review this?

sftim commented:
> @sftim Could you please review this?

I can, but there are two [suggested reviewers](https://github.com/kubernetes/website/pull/49838#event-16362659001) - it's handy if you nudge both.

sftim commented on hugo.toml:
```diff
@@ -334,7 +339,7 @@ languagedirection = "ltr"
 # A list of language codes to look for untranslated content, ordered from left to right.
 language_alternatives = ["en"]
-description = "Solution professionnelle d’orchestration de conteneurs"
+description = "Solution professionnelle d'orchestration de conteneurs"
```
Don't include this change.

karthikmanam replied:
Done

nate-double-u replied:
This has been marked as resolved, but the change still appears to be in the PR.

sftim reviewed (commented):
With one tweak (don't change French), I can give this an LGTM.

karthikmanam commented:
> With one tweak (don't change French), I can give this an LGTM.

@sftim Done. Thanks for your patience and guidance on this.

karthikmanam commented:
@nate-double-u Gentle reminder for your approval.

karthikmanam commented:
> > @sftim Could you please review this?
> 
> I can, but there are two [suggested reviewers](https://github.com/kubernetes/website/pull/49838#event-16362659001) - it's handy if you nudge both.

Acknowledged

sftim commented:
/lgtm

Consider using the actual (butterfly) logo for Bluesky

karthikmanam commented:
> /lgtm
> 
> Consider using the actual (butterfly) logo for Bluesky

@nate-double-u @sftim I searched but that logo is not available

karthikmanam commented:
> > /lgtm
> > Consider using the actual (butterfly) logo for Bluesky
> 
> @nate-double-u @sftim I searched but that logo is not available

The problem is that:
- We updated the icon class in hugo.toml to fa-brands fa-bluesky
- We also updated the local npm package to Font Awesome 6.5.2 which includes the Bluesky icon
- However, the Docker container that's running the Hugo server (gcr.io/k8s-staging-sig-docs/k8s-website-hugo:v0.133.0-1b9242684415) has an older version of Font Awesome baked into it (likely 6.2.0)
- Since the Bluesky icon wasn't added until Font Awesome 6.5.0, the container doesn't recognize the fa-bluesky class and is falling back to displaying a default icon (which appears to be the Twitter logo).

To fix this, we have two options:
- Use a different icon that exists in the older Font Awesome version (like continuing to use fa-cloud or another suitable icon)
- Push our changes (including the updated package.json with Font Awesome 6.5.2) to the repository and wait for the official build process to create a new container with the updated Font Awesome

karthikmanam commented:
> > > /lgtm
> > > Consider using the actual (butterfly) logo for Bluesky
> > 
> > 
> > @nate-double-u @sftim I searched but that logo is not available
> 
> The problem is that:
> 
>     * We updated the icon class in hugo.toml to fa-brands fa-bluesky
> 
>     * We also updated the local npm package to Font Awesome 6.5.2 which includes the Bluesky icon
> 
>     * However, the Docker container that's running the Hugo server (gcr.io/k8s-staging-sig-docs/k8s-website-hugo:v0.133.0-1b9242684415) has an older version of Font Awesome baked into it (likely 6.2.0)
> 
>     * Since the Bluesky icon wasn't added until Font Awesome 6.5.0, the container doesn't recognize the fa-bluesky class and is falling back to displaying a default icon (which appears to be the Twitter logo).
> 
> 
> To fix this, we have two options:
> 
>     * Use a different icon that exists in the older Font Awesome version (like continuing to use fa-cloud or another suitable icon)
> 
>     * Push our changes (including the updated package.json with Font Awesome 6.5.2) to the repository and wait for the official build process to create a new container with the updated Font Awesome

@nate-double-u @sftim I went ahead and created an issue in k8s.io repo. Please see below: 
https://github.com/kubernetes/k8s.io/issues/7884

chris-short commented:
I appreciate the work here but, there's one thing I want to clarify.

The last time I tried to do this:

1. FontAwesome is coming from the Docsy theme itself (which gets pulled in by the container image, on build I believe)
2. We did not want to leave potential breaking changes to the website that aren't resolving the issue at hand.
3. I forgot, or didn't know of, the best place to document my findings so I put them in the issue and/or PR
4. My documentation method did not work as @karthikmanam went down the exact same path as I did

@nate-double-u Where should we put these notes so we know not to try this again?

karthikmanam commented:
@nate-double-u / @sftim Please approve or let me know what path we should take on this PR.

sftim commented:
Just to be clear about what I meant with https://github.com/kubernetes/website/pull/49838#issuecomment-2724569253:

- This PR **looks good to me** and I don't object to it merging, based on the tree (set of files) it had at time time.
- It's even better if we can find a way, maybe later, to use the right icon.

SayakMukhopadhyay commented on hugo.toml:
```diff
@@ -257,12 +257,17 @@ no = 'Sorry to hear that. Please <a href="https://github.com/USERNAME/REPOSITORY
-#	name_key = "community_x_name"
-#	url = "https://twitter.com/kubernetesio"
-#	icon = "fa-brands fa-twitter"
+[[params.links.user]]
+	name_key = "community_x_name"
+	url = "https://twitter.com/kubernetesio"
+	icon = "fa-brands fa-twitter"
```
Is enabling the X icon intended?

karthikmanam replied:
@SayakMukhopadhyay Yes. 
@sftim FYI

SayakMukhopadhyay replied:
Oh, I see his comment now https://github.com/kubernetes/website/pull/49838#discussion_r1963959804

SayakMukhopadhyay commented on hugo.toml:
```diff
@@ -334,7 +339,6 @@ languagedirection = "ltr"
 time_format_blog = "02.01.2006"
 # A list of language codes to look for untranslated content, ordered from left to right.
 language_alternatives = ["en"]
-description = "Solution professionnelle d’orchestration de conteneurs"
```
Is the removal of the description intended?

karthikmanam commented:
> I believe @chris-short had created #49608 which didn't fix the icon issue, even in netlify. I can see that in the netlify preview of the current PR does show the icon correctly, which is a bit weird cause the last time I tried to upgrade fontawesome locally, it didn't work. I will need to do another check with this PR.
> 
> Maybe putting fontawesome in `dependencies` instead of `devDependencies` did the trick.
> 
> Although, @karthikmanam since the change is the addition of a dependency, I suggest making the change in `package.json` and let the lock file generate from the `package.json` else they will get out of sync.

@SayakMukhopadhyay Changing the icon in the package.json is not working. I have documented my findings here: https://github.com/kubernetes/website/pull/49838#issuecomment-2727376828

SayakMukhopadhyay commented:
Oh, I misunderstood. The PR currently doesn't use `fa-bluesky`. I thought it did. But since it doesn't, is there any reason to keep the change to the `package-lock.json`?

shurup commented:
I'm sorry to break in but the Bluesky links are already implemented in this earlier PR: #49606. It takes care of everything, including the logo and some required style changes. It also adds an additional link to the Community page. I think we can close this PR in favour of that one?

karthikmanam commented:
@sftim @nate-double-u Please advise further course of action on this.

nate-double-u commented:
> @nate-double-u Where should we put these notes so we know not to try this again?

I wonder if the style guide would be a good spot, @chris-short?

nate-double-u commented:
Or maybe the [website/wiki](https://github.com/kubernetes/website/wiki)?

nate-double-u commented:
> I'm sorry to break in but the Bluesky links are already implemented in this earlier PR: https://github.com/kubernetes/website/pull/49606. It takes care of everything, including the logo and some required style changes. It also adds an additional link to the Community page. I think we can close this PR in favour of that one?

> @sftim @nate-double-u Please advise further course of action on this.

Since this PR was opened later, and https://github.com/kubernetes/website/pull/49606 seems more complete, I think we should close this one in favour of https://github.com/kubernetes/website/pull/49606.

Thanks for your work on this @karthikmanam, it was helpful to have the discussions here to help us understand what needs to be done.

/close
